src/AppInstallerCommonCore/MSStoreDownload.cpp

```diff
@@ -941,13 +1000,13 @@ namespace AppInstaller::MSStore
             const auto& appContent = appContents.at(0);
 
             // Populate main packages
-            result.MainPackages = PopulateSfsAppFileToMSStoreDownloadFileVector(appContent.GetFiles(), requiredArchitecture, requiredPlatform);
+            result.MainPackages = PopulateSfsAppFileToMSStoreDownloadFileVector(appContent.GetFiles(), requiredArchitecture, requiredPlatform, targetOSVersion);
 
             // Populate dependency packages
             for (auto const& dependencyEntry : appContent.GetPrerequisites())
             {
                 // Not passing in required platform for dependencies. Dependencies are mostly Windows.Universal.
-                auto dependencyPackages = PopulateSfsAppFileToMSStoreDownloadFileVector(dependencyEntry.GetFiles(), requiredArchitecture);
+                auto dependencyPackages = PopulateSfsAppFileToMSStoreDownloadFileVector(dependencyEntry.GetFiles(), requiredArchitecture, Manifest::PlatformEnum::Unknown, targetOSVersion);

```

Comment:
Is it possible for dependency packages to have higher min OS version than main package? Maybe we can pass nullopt for targetOSVersion?